Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs

```diff
@@ -1579,8 +1611,17 @@ public override void Update(float elapsedClockSeconds)
 
             if (!AdvancedAdhesionModel)  // Advanced adhesion model turned off.
                AbsWheelSpeedMpS = AbsSpeedMpS;
-
-            UpdateMotiveForce(elapsedClockSeconds, t, AbsSpeedMpS, AbsWheelSpeedMpS);
+            // Jindrich
+            //UpdateMotiveForce(elapsedClockSeconds, t, AbsSpeedMpS, AbsWheelSpeedMpS);
+            if (CruiseControl != null)
+            {
+                if (!IsPlayerTrain || CruiseControl.SpeedRegMode == CruiseControl.SpeedRegulatorMode.Manual || CruiseControl.UseThrottle)
+                    UpdateMotiveForce(elapsedClockSeconds, t, AbsSpeedMpS, AbsWheelSpeedMpS);
+                else if (IsPlayerTrain || CruiseControl.SpeedRegulatorOptions.Contains("engageforceonnonzerospeed"))
+                    CruiseControl.Update(elapsedClockSeconds, AbsWheelSpeedMpS);
```

Comment:
Please have a look at this lines:
If train is not a player train, the first clause will be executed, which I believe it's OK
However, if train is player train, the 'else if' condition is unnecessary, so it could be replaced with an 'else'. Did you mean something else?

Reply:
Fixed.